wger/core/api/views.py

```diff
@@ -108,3 +110,53 @@ class WeightUnitViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = WeightUnitSerializer
     ordering_fields = '__all__'
     filter_fields = ('name', )
+
+class UserRegistrationView(viewsets.ModelViewSet):
+    '''
+    API endpoint for user registration
+    '''
+    def get_serializer_class(self, request=None):
+        if request:
+            return ApiUserSerializer
+        return UserSerializer
+
+    def create(self, request):
```

Comment:
Your creating the user before verifying whether the account/ app that is creating the account is allowed to do so. User descriptive names for your variables. Consider breaking these functions into two different methods(Create user, and Create API user)